src/transformers/trainer.py

```diff
@@ -2101,7 +2101,13 @@ def _load_from_checkpoint(self, resume_from_checkpoint, model=None):
                     # release memory
                     del state_dict
             elif self.is_fsdp_enabled:
-                load_fsdp_model(self.accelerator.state.fsdp_plugin, self.accelerator, model, resume_from_checkpoint)
+                load_fsdp_model(
```

Comment:
You need to first check if `adapter_only` is correctly in `load_fsdp_model` signature, and if that's the case, pass `adapter_only=True`, if not, maybe throw a warning message with `logger.warning()` advising users to upgrade accelerate to be able to properly save adapters with FSDP. 
You can check a simple example on how to check if an arg is in a method signature for example here: https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L746 - ideally we want to check that once and not multiple times

Reply:
Thanks @younesbelkada I was imagining the `accelerate` PR is merged before this one. But that works too.

Reply:
yes but I meant for compatibility for users that have previous versions of accelerate. If I have accelerate==0.21.0 and transformers with your patch I'll get an error because you always pass `adapter_only` to `load_fsdp_model`, does that makes sense?

Reply:
Yes, sorry, I meant I was imagining the dependency requirement would get bumped to the latest accelerate, but I'll work on inspecting the function signature on this PR, I think that might be a better solution to keep backwards compatibility.

Reply:
ahh I see, no usually we don't bump libraries version to ensure BC, indeed it shouldn't be that hard to inspect the method signature, let me know if you need any help !

Reply:
This is now done @younesbelkada. Thanks for the pointer. I've also tested it with the `branch_test.zip` (that pulled in the latest changes from this PR and the changes in the `accelerate` PR).

Reply:
perfect, thanks!